mat64/symmetric.go

```diff
@@ -5,16 +5,41 @@ import (
 	"github.com/gonum/blas/blas64"
 )
 
-// Symmetric represents a symmetric matrix.
-type Symmetric struct {
+var (
+	symDense *SymDense
+
+	_ Matrix          = symDense
+	_ Symmetric       = symDense
+	_ RawSymmmetricer = symDense
+)
+
+const (
+	ErrUplo = "mat64: blas64.Symmetric not upper"
+)
+
+// SymDense is a symmetric matrix that uses Dense storage.
+type SymDense struct {
 	mat blas64.Symmetric
 }
 
-// NewSymmetric constructs an n x n symmetric matrix. If len(mat) == n * n,
+// Symmetric represents a symmetric matrix (where the element at {i, j} equals
+// the element at {j, i}). Symmetric matrices are always square.
+type Symmetric interface {
+	Matrix
+	// Symmetric returns the number of rows/columns in the matrix.
+	Symmetric() int
```

Comment:
It shouldn't be that general. The interface for Triangular will also heed a similar third function. Triangular should not implement Symmetric and vis-versa.